ui/src/services/alertUtils.js

```diff
@@ -0,0 +1,208 @@
+  return alerts.map((alert) => {
+    let parents = [];
+    let selectors = [];
+
+    if (isObjectHasProperty(alert.labels, 'parents')) {
+      parents.push(alert.labels.parents.split(','));
+    }
+    if (isObjectHasProperty(alert.labels, 'selectors')) {
+      selectors.push(alert.labels.selectors.split(','));
+    }
+    return {
+      summary: alert.annotations.summary,
+      description: alert.annotations.description || alert.annotations.message,
+      startsAt: alert.startsAt,
+      endsAt: alert.endsAt || new Date().toISOString(),
+      severity: alert.labels.severity,
+      documentationUrl: alert.annotations.runbook_url,
+      labels: { ...alert.labels, parents, selectors },
+    };
+  });
+};
+
+/* 
+This function is used when filtering the alerts, by checking if the `labels` is 
+the subset of the `filters` object. 
+Note that: The values of the selectors can be more than one, so it's possible to be an array.
+
+e.g. alertname: [NODE_FILESYSTEM_SPACE_FILLINGUP, NODE_FILESYSTEM_ALMOST_OUTOF_SPACE, 
+NODE_FILESYSTEM_FILES_FILLINGUP, NODE_FILESYSTEM_ALMOST_OUTOF_FILES]  
+*/
+export const isAlertSelected = (
+  labels: {
+    [labelName: string]: string,
+    parents?: string[],
+    selectors?: string[],
+  },
+  filters: {
+    [labelName: string]: string | string[],
+    parents?: string[],
+    selectors?: string[],
+  },
+): boolean => {
+  return Object.getOwnPropertyNames(filters)
+    .map((key) => {
+      if (!filters[key] || !labels[key] || !isObjectHasProperty(labels, key))
+        return false;
+      if (Array.isArray(filters[key])) {
+        if (Array.isArray(labels[key])) {
+          return (
+            filters[key].find((val) => {
+              return labels[key].indexOf(val) >= 0;
+            }) !== undefined
+          );
+        } else return filters[key].includes(labels[key]);
+      } else {
+        return labels[key].includes(filters[key]);
+      }
+    })
+    .reduce((boolA, boolB) => {
+      return boolA && boolB;
+    });
+};
+
+// TODO: Call Loki Endpoint
+export const getHistoryAlerts = (
+  start: string, //TODO: maybe default last 7 days
+  end?: string = new Date().toISOString(),
+  filters: {
+    [labelName: string]: string | string[],
+    parents?: string[],
+    selectors?: string[],
+  },
+): AlertList => {
+  return getAlertsLoki(start, end).filter((alert) => {
+    return isAlertSelected(alert.labels, filters);
+  });
+};
+
+// Call Alertmanager Endpoint
+export const getActiveAlerts = (
+  alerts: AlertList,
+  filters: {
+    [labelName: string]: string | string[],
+    parents?: string[],
+    selectors?: string[],
+  },
+): AlertList => {
+  return alerts.filter((alert) => {
+    return isAlertSelected(alert.labels, filters);
+  });
+};
+
+// check if the given time is DURING the start and end or OUT.
+export const firingTime = (
+  start: string,
+  end: string,
+  on: string,
+): 'OUT' | 'DURING' => {
```

Comment:
You may want to represent the output of this function using a boolean type and rename your function to `dateIsBetween(date: string, start: string, end: string): boolean`